irohad/validation/impl/stateful_validator_impl.cpp

```diff
@@ -29,34 +33,88 @@ namespace iroha {
       log_ = logger::log("SFV");
     }
 
-    std::shared_ptr<shared_model::interface::Proposal>
-    StatefulValidatorImpl::validate(
+    validation::VerifiedProposalAndErrors StatefulValidatorImpl::validate(
         const shared_model::interface::Proposal &proposal,
         ametsuchi::TemporaryWsv &temporaryWsv) {
       log_->info("transactions in proposal: {}",
                  proposal.transactions().size());
-      auto checking_transaction = [](const auto &tx, auto &queries) {
-        return bool(queries.getAccount(tx.creatorAccountId()) |
-                    [&](const auto &account) {
-                      // Check if tx creator has account and has quorum to
-                      // execute transaction
-                      return boost::size(tx.signatures()) >= account->quorum()
-                          ? queries.getSignatories(tx.creatorAccountId())
-                          : boost::none;
-                    }
-                    |
-                    [&](const auto &signatories) {
-                      // Check if signatures in transaction are account
-                      // signatory
-                      return signaturesSubset(tx.signatures(), signatories)
-                          ? boost::make_optional(signatories)
-                          : boost::none;
-                    });
+      auto checking_transaction = [this](const auto &tx, auto &queries) {
+        return expected::Result<void, validation::CommandNameAndError>(
+            [&]() -> expected::Result<
+                         std::shared_ptr<shared_model::interface::Account>,
+                         validation::CommandNameAndError> {
+              // Check if tx creator has account
+              auto account = queries.getAccount(tx.creatorAccountId());
+              if (account) {
+                return expected::makeValue(*account);
+              }
+              return expected::makeError(validation::CommandNameAndError{
+                  "Initial transaction verification: no such account",
+                  (boost::format("stateful validator error: could not fetch "
+                                 "account with id %s")
+                   % tx.creatorAccountId())
+                      .str()});
+            }() |
+                [&](const auto &account)
```

Comment:
would be nice to move out this labmdas, so the code will be more readable

Reply:
Isn't it? From my point of view, those lambdas form a logical pipeline, which is very nice...